go/parquet/internal/testutils/random.go

```diff
@@ -369,6 +370,18 @@ func randFloat64(r *rand.Rand) float64 {
 	}
 }
 
+// randFloat16 creates a random float value with a normal distribution
+// to better spread the values out and ensure we do not return any NaN or Inf values.
+func randFloat16(r *rand.Rand) float16.Num {
+	for {
+		f16 := float16.FromBits(uint16(r.Uint64n(math.MaxUint16 + 1)))
+		f64 := float64(f16.Float32())
+		if !math.IsNaN(f64) && !math.IsInf(f64, 0) {
```

Comment:
It might make more sense to leave the NaN's and infinities to ensure we're covering those cases for testing. Alternately we could just make sure we test those cases separately